frontend/packages/console-plugin-sdk/src/typings/features.ts

```diff
@@ -4,42 +4,34 @@ import { Extension, AlwaysOnExtension } from './base';
 import { Action } from 'typesafe-actions';
 
 namespace ExtensionProperties {
-  interface FeatureFlag {
-    /** The name of the feature flag. */
+  export interface ModelFeatureFlag {
     flag: string;
-  }
-
-  export interface ModelFeatureFlag extends FeatureFlag {
     /** If a CRD for this model exists, the feature will be enabled. */
     model: K8sKind;
   }
 
-  export interface ActionFeatureFlag extends FeatureFlag {
-    /** Function used to detect the feature and set flag name/value via Redux action dispatch. */
-    detect: ActionFeatureFlagDetector;
+  export interface FeatureDetect {
```

Comment:
Let's follow the existing naming convention.

Extension property type should have the same name (namespaced under `ExtensionProperties`) as the corresponding extension type.

Furthermore, since the goal here is to support detection & setting of arbitrary feature flags, I'd name it `CustomFeatureFlag`.

```ts
interface CustomFeatureFlag {
  /** Detect the feature and set the corresponding flag name/value via Redux action dispatch. */
  detect: FeatureDetector;
}
```

```ts
export interface CustomFeatureFlag extends Extension<ExtensionProperties.CustomFeatureFlag> {
  type: 'FeatureFlag/Custom';
}

export const isCustomFeatureFlag = (e: Extension): e is CustomFeatureFlag => {
  return e.type === 'FeatureFlag/Custom';
};
```